NitroxPatcher/Patches/Dynamic/uGUI_SceneIntro_IntroSequence_Patch.cs

```diff
@@ -0,0 +1,187 @@
+    }
+
+    private static float GetSkipTime()
+    {
+        // Return Time.time when starting solo and disable skip button when staring duo
+        return Resolve<LocalPlayer>().IntroCinematicMode == IntroCinematicMode.SINGLEPLAYER ? Time.time : -1f;
+    }
+
+    // ReSharper disable once UnusedMethodReturnValue.Local
+    private static bool IsRemoteCinematicReady(uGUI_SceneIntro uGuiSceneIntro)
+    {
+        if (callbackRun) return true;
+
+        if (GameModeUtils.currentGameMode.HasFlag(GameModeOption.Creative)) uGuiSceneIntro.Stop(true); // Stopping intro if Creative like in normal SN
+
+        if (Resolve<LocalPlayer>().IntroCinematicMode == IntroCinematicMode.COMPLETED)
+        {
+            uGuiSceneIntro.Stop(true);
+            EndRemoteCinematic();
+            return false;
+        }
+
+        if (!packetSend)
+        {
+            uGuiSceneIntro.skipHintStartTime = Time.time;
+            uGuiSceneIntro.mainText.SetText(Language.main.GetFormat("Nitrox_IntroWaitingPartner", uGUI.FormatButton(GameInput.Button.UIMenu)));
+            uGuiSceneIntro.mainText.SetState(true);
+
+            Resolve<PlayerCinematics>().SetLocalIntroCinematicMode(IntroCinematicMode.WAITING);
+            packetSend = true;
+            IsWaitingForPartner = true;
+            return false;
+        }
+
+        ushort? opPartnerId = Resolve<PlayerCinematics>().IntroCinematicPartnerId;
+
+        if (Resolve<LocalPlayer>().IntroCinematicMode == IntroCinematicMode.START &&
+            opPartnerId.HasValue && Resolve<PlayerManager>().TryFind(opPartnerId.Value, out RemotePlayer newPartner))
+        {
+            partner = newPartner;
+            EnqueueStartCinematic(uGuiSceneIntro);
+            Resolve<PlayerCinematics>().SetLocalIntroCinematicMode(IntroCinematicMode.PLAYING);
+            Resolve<PlayerCinematics>().IntroCinematicPartnerId = null;
+        }
+
+        return false;
+    }
+
+    public static void EnqueueStartCinematic(uGUI_SceneIntro uGuiSceneIntro)
+    {
+        IsWaitingForPartner = false;
+        uGuiSceneIntro.skipHintStartTime = -1;
+        uGuiSceneIntro.moveNext = false;
+        uGuiSceneIntro.mainText.FadeOut(0.2f, uGuiSceneIntro.Callback);
+        callbackRun = true;
+    }
+
+    private static void StartRemoteCinematic()
+    {
+        if (partner != null) // Is null when RunCinematicSingleplayer() is called
+        {
+            partner.PlayerModel.transform.localScale = new Vector3(-1, 1, 1);
+            partner.ArmsController.enabled = false;
+            partner.AnimationController.UpdatePlayerAnimations = false;
+            partner.AnimationController["cinematics_enabled"] = true;
+            partner.AnimationController["escapepod_intro"] = true;
+
+            IntroCinematicUpdater.Partner = partner;
+            partner.Body.AddComponent<IntroCinematicUpdater>();
+        }
+    }
+
+    private static void EndRemoteCinematic()
+    {
+        if (partner != null)
```

Comment:
"partner" is not null even when the partner disconnects to fix that we can either:
- replace: ``partner != null`` by ``IsPartnerValid()`` in the two methods
```cs
private static bool IsPartnerValid()
{
    return partner != null && Resolve<PlayerManager>().Find(partner.PlayerId).HasValue;
}
```
- have the patch listen to ``PlayerManager.OnRemove``, which might be a bit too much pain to find the right time to register and then to unregister the callback